dlib/matrix/matrix_math_functions_abstract.h

```diff
@@ -587,6 +587,22 @@ namespace dlib
                   R(r,c) == std::tanh(m(r,c)) 
     !*/
 
+// ----------------------------------------------------------------------------------------
+
+    const matrix_exp soft_max (
+        const matrix_exp& m
+    );
+    /*!
+        requires
+            - matrix_exp::type == float, double, or long double
+        ensures
+            - returns a matrix R such that:
+                - R::type == the same type that was in m
+                - R has the same dimensions as m
+                - for all valid r and c:
+                  R(r,c) == std::exp(m(r,c) / sum(std::exp(m))
```

Comment:
There is a typo here.  It should be std::exp(m(r,c)) / sum(std::exp(m))

Reply:
fixed! :)